reporter/http/http.go

```diff
@@ -80,24 +80,34 @@ func (r *httpReporter) loop() {
 			currentBatchSize := r.append(span)
 			if currentBatchSize >= r.batchSize {
 				nextSend = time.Now().Add(r.batchInterval)
-				go func() {
-					_ = r.sendBatch()
-				}()
+				r.enqueueSend()
 			}
 		case <-tickerChan:
 			if time.Now().After(nextSend) {
 				nextSend = time.Now().Add(r.batchInterval)
-				go func() {
-					_ = r.sendBatch()
-				}()
+				r.enqueueSend()
 			}
 		case <-r.quit:
-			r.shutdown <- r.sendBatch()
+			close(r.sendC)
 			return
 		}
 	}
 }
 
+func (r *httpReporter) sendLoop() {
+	for range r.sendC {
+		_ = r.sendBatch()
+	}
+	r.shutdown <- r.sendBatch()
+}
+
+func (r *httpReporter) enqueueSend() {
+	select {
+	case r.sendC <- struct{}{}:
+	default:
```

Comment:
Not sure it is a good idea to have a `default` with empty body. It might have intensive CPU usage (unless I am missing something here). Also, maybe we just need a `r.sendC <- struct{}{}`?

Reply:
The default case will be hit only if we're enqueueing a new send request and there's one pending send request in the `sendC` channel already. In this case we wouldn't want to block the `loop()` goroutine to wait for the pending send request to get processed since that would also block `Send()` users waiting to send on the `spanC` channel. Maybe a comment in the `default:` case would make things a bit clearer?

The current rate limiting logic in `loop()` wasn't changed by this patch.